### PR TITLE
Feature/job restart policy options #129

### PR DESCRIPTION
**Pull request by robert-clegg-tessella**
Stories covered: #129 

Settings for restarting Spark jobs are configurable in `validation_rules.json`:
* `onFailureRetries`
* `onFailureRetryInterval`
* `onSubmissionFailureRetries`
* `onSubmissionFailureRetryInterval`

Also in this PR:
* System tests check for status more frequently to avoid missing completed jobs before they are cleaned up
* python2 script for cleaning up old Docker images

Note also the changes to the wiki guidance on setting [`TidyFrequency`](https://github.com/ukaea/piezo/wiki/WebAppDeployment#tidy-frequency)

### CODE REVIEW

File: piezo_web_app/PiezoWebApp/tests/integration_tests/submit_job_test.py [hunk 1]
```diff
@@ -71,7 +71,13 @@ def test_correct_python_job_is_submitted_correctly(self):
                 'imagePullPolicy': 'Always',
                 'mainApplicationFile': '/path_to/file',
                 'sparkVersion': '2.4.0',
-                'restartPolicy': {'type': 'Never'},
+                'restartPolicy': {
+                    'type': 'Never',
```
Comment (oliver-tarrant-tessella) [inline piezo_web_app/PiezoWebApp/tests/integration_tests/submit_job_test.py]: Just for clarity `Never` should probably be changed to `onFailure`

Same for following two

File: piezo_web_app/PiezoWebApp/tests/services/spark_job/validation/manifest_populator_test.py [hunk 2]
```diff
@@ -26,6 +26,10 @@ def setup(self):
                 'image_pull_policy': 'Always',
                 'spark_version': '2.4.0',
                 'restart_policy': 'Never',
```
Comment (oliver-tarrant-tessella) [inline piezo_web_app/PiezoWebApp/tests/services/spark_job/validation/manifest_populator_test.py]: As above

File: piezo_web_app/PiezoWebApp/tests/integration_tests/submit_job_test.py (hunk 1 above)
Comment (robert-clegg-tessella) [reply piezo_web_app/PiezoWebApp/tests/integration_tests/submit_job_test.py]: 11aadbd

File: piezo_web_app/PiezoWebApp/tests/services/spark_job/validation/manifest_populator_test.py (hunk 2 above)
Comment (robert-clegg-tessella) [reply piezo_web_app/PiezoWebApp/tests/services/spark_job/validation/manifest_populator_test.py]: 11aadbd

**Pull request merged**